packages/realm/src/OrderedCollection.ts

```diff
@@ -360,11 +374,17 @@ export abstract class OrderedCollection<T = unknown, EntryType extends [unknown,
    */
   indexOf(searchElement: T, fromIndex?: number): number {
     assert(typeof fromIndex === "undefined", "The second fromIndex argument is not yet supported");
+
     if (this.type === "object") {
       assert.instanceOf(searchElement, RealmObject);
-      return this.results.indexOfObj(searchElement[INTERNAL]);
+      return this.results.indexOfObj(searchElement[OBJ_INTERNAL]);
+    } else if (isJsOrRealmList(searchElement) || isJsOrRealmDictionary(searchElement)) {
+      // Collections are always treated as not equal since their
+      // references will always be different for each access.
+      const NOT_FOUND = -1;
+      return NOT_FOUND;
```

Comment:
Why do we use a const here?

Reply:
Do you mean why not just return `-1`? It's merely for readability.

Reply:
I understand you want to make clear that -1 is used when the value is not found, but for me it looks a little bit strange creating a const variable and then using it straight away in the same function. Aren't we also creating a new variable unnecessarily every time the method is called?
I think a comment would suffice in this case, or maybe move the const declaration at a class or higher level. 
Just an observation though, I don't have super strong opinion on this.

Reply:
I see your point 🙂 In this case, this code only runs when the value to search for is a collection, and these two statements would likely just be replaced with returning `-1` in the end anyway. In this context I think just having `-1` (without either the variable or a comment) would not be too bad, but I think the readability increases if you do have it.